### ISSUE TITLE
NET-1648 Fix S1451 FN: Razor files

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8350]
Originated from [this](https://github.com/SonarSource/sonar-dotnet/pull/8341#discussion_r1388016467) discussion.

Razor and cshtml files have multiple ways of defining comments:

```cs
<!-- Copyright disclaimer as an HTML Comment --> 
@* Copyright disclaimer as a Razor comment *@
@{/* Copyright disclaimer as a C# multiline comment */}
@{ // Copyright disclaimer as a C# single-line comment
}
```

None of these is currently supported by S1451 on `razor` files. 
As for `cshtml` files, unlike `razor` files, both the .NET and the HTML analyzers are executed, so the HTML flavor of S1451 captures the `<!-- -->` cases, but none of the others, that are specific to Razor. That means that the rule would need to raise for `razor` files, but shouldn't raise on `cshtml` files. Otherwise, two different analyzers would report the same issue.
Moreover, at the time of this writing, the current way of analyzing Razor code is via the generated source, which may not include line directives for pure Razor-only code, such as `@* *@`, or for HTML code, such as `<!-- -->`.

So particular care has to be put into assessing the feasibility of this task. 
Two possibilities to explore are:
- check the `razor` source at the text level, and don't count on the AST from Roslyn
- wait for Roslyn to integrate razor templating into parsing and compilation

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1648](https://sonarsource.atlassian.net/browse/NET-1648)

[NET-1648]: https://sonarsource.atlassian.net/browse/NET-1648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
